File: modules/spring-core/testdata/java/inspection/finalDeclaration/src/com/GeneratorServiceImpl.java
package com;

import org.springframework.cache.annotation.Caching;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.Async;
import org.springframework.transaction.annotation.Propagation;

import javax.swing.text.html.parser.Entity;

@Async
public final class GeneratorServiceImpl implements GeneratorService {
    @Override
    public void generatorService(final Entity entity) {}

    @org.springframework.transaction.annotation.Transactional
    public final void key(){}
}

interface GeneratorService {
    void generatorService(Entity entity);
}

class CachingService{
    @Caching
    public final void caching(){
    }
}

@Configuration
class FinalConfiguration {
    @Bean
    final FinalObject getObj() {
        return new FinalObject();
    }
}

class FinalObject {}

File: modules/spring-core/testdata/kotlin/inspection/finalDeclaration/src/com/GeneratorServiceImpl.kt
package com

import org.springframework.cache.annotation.Caching
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.Async
import org.springframework.transaction.annotation.Transactional
import javax.swing.text.html.parser.Entity

@Async
class GeneratorServiceImpl : GeneratorService {
    override fun generatorService(entity: Entity?) {}

    @Transactional
    fun key() {
    }
}

internal interface GeneratorService {
    fun generatorService(entity: Entity?)
}

open class CachingService {
    @Caching
    fun caching() {
    }
}

@Configuration
open class FinalConfiguration {
    @Bean
    fun getObj(): FinalObject {
        return FinalObject()
    }
}

class FinalObject